lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -814,21 +837,20 @@ def process(self, historic_forecast, truth):
         forecast_var = historic_forecast.collapsed(
             "realization", iris.analysis.VARIANCE)
 
+        # If a landsea_mask is provided mask out the sea points
+        if landsea_mask:
+            self.mask_cube(forecast_predictor, landsea_mask)
+            self.mask_cube(forecast_var, landsea_mask)
+            self.mask_cube(truth, landsea_mask)
+
         # Computing initial guess for EMOS coefficients
-        # If no initial guess from a previous iteration, or if there
-        # are NaNs in the initial guess, calculate an initial guess.
-        if "initial_guess" not in locals() or nan_in_initial_guess:
-            initial_guess = self.compute_initial_guess(
-                truth, forecast_predictor, self.predictor_of_mean_flag,
-                self.ESTIMATE_COEFFICIENTS_FROM_LINEAR_MODEL_FLAG,
-                no_of_realizations=no_of_realizations)
-
-        if np.any(np.isnan(initial_guess)):
-            nan_in_initial_guess = True
-
-        if not nan_in_initial_guess:
-            # Need to access the x attribute returned by the
-            # minimisation function.
+        initial_guess = self.compute_initial_guess(
```

Comment:
The definition of the `nan_in_initial_guess` variable can now be removed as well.

Reply:
Done (surprised I missed that)